daffodil-tdml-processor/src/main/scala/org/apache/daffodil/tdml/processor/DaffodilTDMLDFDLProcessor.scala

```diff
@@ -159,14 +163,18 @@ final class TDMLDFDLProcessorFactory private (
   override def getProcessor(
     schemaSource: DaffodilSchemaSource,
     useSerializedProcessor: Boolean,
-    optRootName: Option[String] = None,
-    optRootNamespace: Option[String] = None): TDML.CompileResult = {
+    optRootName: Option[String],
+    optRootNamespace: Option[String],
+    tunables: Map[String, String]): TDML.CompileResult = {
```

Comment:
Why pass in tunbles now? Doesn't the TDML runner call setTunables?

Reply:
Some of the tunables affect compilation, hence, unless they are passed in, the cache can return a false match. The unqualifiedPathStepPolicy is the primary one where I was actually getting a test failure (can't recall which test), but there can (perhaps are) others, so I decided to just pass them all in. 
We could make a subset data structure with just the compler-affecting tunables in it, but there are a lot of them, many are used both at compile time and runtime. 

These are the tunables accessed by schema compilation:
unqualifiedPathStepPolicy
requireTextStandardBaseProperty
requireEmptyElementParsePolicyProperty
allowExpressionResultCoercion
requireChoiceDispatchKeyKindProperty
requireFloatingProperty
generatedNamespacePrefixStem
requireTextBidiProperty
requireEncodingErrorPolicyProperty
maxBinaryDecimalVirtualPoint
minBinaryDecimalVirtualPoint
parseUnparsePolicy
maxSkipLengthInBytes
initialRegexMatchLimitInCharacters
defaultEmptyElementParsePolicy
... and any used by expressions which are evaluated at compile time for constant folding purposes.
e.g., maxValidYear, minValidYear, 

Any of these changing *could* result in an SDE vs. successful compilation.

Reply:
That seems too difficult and hard to maintain. I guess it's not a big deal.

A thought though, I wonder if maybe the caching shouldn't happen in the DaffodilTDMLDFDLProcessor? And it should be up to the TDMLRunner to cache things? That way we can cache any processor, regardless of the implementation and implmentations don't have to worry about it?

Though, might not be worth the effort since I think there are some thoughts about removing caching entirely?

Reply:
The caching needs to get hoisted out of DaffodilTDMLDFDLProcessor into TDMLRunner in order for it to not be a global cache, but a per TestSuite object cache. We do need some caching to avoid recompilation for every test in a test suite.

Reply:
This is beyond the scope of this change, but will we even need the cache at all? Could our test suites become something like this:
```scala
object TestFoo {
  var runner: Runner = _
  @BeforeClass def setup: Unit = { runner = Runner("foo.tdml") }
  @AfterClass def tearDown: Unit = { runner = null }
}
```
This way we compile the TMDL files when JUnit is about to start the tests, keep it around for the life of the suite, and then set it to null once all tests are complete and let the garbage collector cleanup the runner?

Reply:
That will work if the cache is part of the testsuite/runner object, not a global singleton.